### PR TITLE
RELATED: RAIL-4093 better API for elements specification in attribute elements query

### PR DESCRIPTION
**Pull request by no23reason**
Propose a new configuration option to specify attribute elements in attribute elements query.
This allows backends that support it to specify what kind of element value they are sending.

Also deprecate the old way of doing the same thing and deprecate not well documented and seemingly unused prompt config.

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7341/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/177/ : UNSTABLE in 13m 04s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7342/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9193/ : SUCCESS in 19m 02s

**Comment by no23reason** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7347/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/178/ : SUCCESS in 12m 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7346/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9198/ : SUCCESS in 16m 24s

File: libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts [hunk 1]
```diff
@@ -20,13 +69,16 @@ export interface IElementsQueryOptions {
...
      * TODO what is this doing?
+     * @deprecated do not use.
      */
     prompt?: string;
 
     /**
-     *  With this option you can specify concrete attribute elements uris to load.
-     *  This is commonly used to preload selected elements in the attribute filter
+     * With this option you can specify concrete attribute elements uris to load.
+     * This is commonly used to preload selected elements in the attribute filter.
```
Comment (plavemar) [inline libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts:80]: The second line of the comments should be in the remarks tag, it says some detailed info about the property

File: libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts [hunk 2]
```diff
@@ -39,6 +91,12 @@ export interface IElementsQueryOptions {
      * Include the total count of all elements in the response (without filters applied)
      */
     includeTotalCountWithoutFilters?: boolean;
+
+    /**
+     * Specify particular elements to load.
+     * This is commonly used to preload selected elements in the attribute filter.
```
Comment (plavemar) [inline libs/sdk-backend-spi/src/workspace/attributes/elements/index.ts:97]: The second line of the comments should be in the remarks tag, it says some detailed info about the property

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7352/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9201/ : SUCCESS in 16m 27s

**Review by ivanmjartan — APPROVED** [review]

**Comment by ivanmjartan** [comment]
LGTM

**Review by ivanmjartan — APPROVED** [review]

**Comment by ivanmjartan** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7355/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9202/ : SUCCESS in 18m 32s

**Comment by gdgate** [comment]
Github ref error: SHA of pull request head checked out from remote PR head ref is different from head commit SHA from Github webhook event.

**Comment by no23reason** [comment]
extended check cypress

**Comment by no23reason** [comment]
extended check cypress

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7362/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tests/181/ : SUCCESS in 10m 06s

**Review by plavemar — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2367

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2124/ : SUCCESS in 1m 19s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2408/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1228/ : SUCCESS in 13m 21s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9210/ : SUCCESS in 16m 34s

**Pull request merged**